commcare/src/test/java/org/motechproject/commcare/service/impl/parser/CommcareEventParsersTest.java

```diff
@@ -163,4 +169,22 @@ public void shouldParseCaseEventParametersProperly() {
         assertEquals("yes", parsedParameters.get("dob_known"));
         assertEquals("1990-09-09", parsedParameters.get("dob_calc"));
     }
+
+    @Test
+    public void shouldParseFormsEventAttributesProperly() {
+        String eventSubject = formsEvent.getSubject();
+        Map<String, Object> eventParameters = formsEvent.getParameters();
+
+        Map<String, Object> parsedParameters = formsEventParser.parseEventParameters(eventSubject, eventParameters);
+
+        assertTrue(parsedParameters.containsKey("/data/case/@case_id"));
+        assertTrue(parsedParameters.containsKey("/data/case/@user_id"));
+        assertTrue(parsedParameters.containsKey("/data/case/@date_modified"));
+        assertTrue(parsedParameters.containsKey("/data/case/@xmlns"));
+
+        assertEquals(CASE_ID, parsedParameters.get("/data/case/@case_id"));
+        assertEquals(USER_ID, parsedParameters.get("/data/case/@user_id"));
+        assertEquals(DATE_MODIFIED, parsedParameters.get("/data/case/@date_modified"));
+        assertEquals(XMLNS, parsedParameters.get("/data/case/@xmlns"));
```

Comment:
Let's modify the same sample XML, to have attributes at varying levels of the XML tree